common/reflect_resource.go

```diff
@@ -629,8 +639,16 @@ func collectionToMaps(v any, s *schema.Schema, aliases map[string]map[string]str
 					return fmt.Errorf("%s: %v", path, err)
 				}
 				data[fieldName] = nv
+			case schema.TypeBool, schema.TypeInt:
+				// If the field is not in the force send fields, we should not store it in the state
+				// Only bool and int values are in force send fields now
+				if len(forceSendFields) > 0 && !slices.Contains(forceSendFields, valueField.sf.Name) {
```

Comment:
We do need to store it in state if it is not a zero value though. I figured we could handle this by adding a case to `isValueNilOrEmpty` since the treatment is the same as that case.